app/components/Basket.js

```diff
@@ -1,21 +1,61 @@
-"use client";
-
 import { useContext } from "react";
-
 import { Context, DispatchContext } from "../Context";
 
 export default function Basket() {
   const { customer } = useContext(Context);
   const dispatch = useContext(DispatchContext);
 
-  const removeDrink = (index) => {
-    dispatch({
-      type: "REMOVE_DRINK",
-      payload: index,
+  const separateDrinks = () => {
+    const drinkCount = {};
+
+    customer.drinks.forEach((drink) => {
+      drinkCount[drink] = (drinkCount[drink] || 0) + 1;
```

Comment:
Consider changing the type of drinkCount to an int instead of an object. Or alternatively change the name of the variable. 

If something is called count it would naturally specify that its a number.